contributing/core/testing.md

```diff
@@ -0,0 +1,64 @@
+# Testing
+
+## Running tests
+
+We recommend running the tests in headless mode (with the browser windows hidden) and with a specific directory pattern and/or test name (`-t`) which ensures only a small part of the test suite is run locally:
+
+For example, running one test in the production test suite:
+
+````
+Running one test in the `test/integration/production` test suite:
+
+```sh
+pnpm test -- test/integration/production -t "should allow etag header support"
+````
+
+Running all tests in the `test/integration/production` test suite:
+
+```sh
+pnpm test test/integration/production
+```
+
+When you want to debug a particular test you can replace `pnpm test` with `pnpm testonly` to opt out of the headless browser.
+When the test runs it will open the browser that is in the background by default, allowing you to inspect what is on the screen.
+
+```sh
+pnpm testonly -- test/integration/production -t "should allow etag header support"
```

Comment:
https://github.com/vercel/next.js/pull/40515/files#r971893479

Reply:
Wow, I didn't notice that before!

I always use commands like this:

```sh
pnpm run testonly --test-path-pattern test/integration/script-loader/test/index.test.js -t "onReady"
```

And pnpm is still able to forward `--test-path-pattern` and `-t` to jest.

Reply:
Jest [docs says](https://jestjs.io/docs/cli#--testpathpatternregex) it's called `--testPathPattern`, and Tim originally recommended it here:

https://github.com/vercel/next.js/pull/39778/files#diff-5d428d569033a870db648131d8e7ffdb602f395e1ee5cedb3821d8145e33c9b0R15

But it did not work for me: 
![image](https://user-images.githubusercontent.com/18369201/190403235-c9306aad-5b69-451d-8241-09f6d80046b4.png)

(It's specific to `pnpm test`, `pnpm testonly` seems to work.)